runtime/cudaq/platform/default/rest_server/helpers/RestRemoteClient.cpp

```diff
@@ -218,6 +254,304 @@ class RemoteRestRuntimeClient : public cudaq::RemoteRuntimeClient {
+  std::string nvcfInvocationUrl() const {
+    return fmt::format("https://{}/nvcf/exec/functions/{}/versions/{}",
+                       m_baseUrl, m_functionId, m_functionVersionId);
+  }
+  // Return the URL to request an Asset upload link
+  std::string nvcfAssetUrl() const {
+    return fmt::format("https://{}/nvcf/assets", m_baseUrl);
+  }
+  // Return the URL to retrieve status/result of an NVCF request.
+  std::string
+  nvcfInvocationStatus(const std::string &invocationRequestId) const {
+    return fmt::format("https://{}/nvcf/exec/status/{}", m_baseUrl,
+                       invocationRequestId);
+  }
+  // Construct the REST headers for calling NVCF REST APIs
+  std::map<std::string, std::string> getHeaders() const {
+    std::map<std::string, std::string> header{
+        {"Authorization", fmt::format("Bearer {}", m_apiKey)},
+        {"Content-type", "application/json"}};
+    return header;
+  };
+  // Helper to retrieve the list of all available versions of the specified
+  // function Id.
+  std::vector<cudaq::NvcfFunctionVersionInfo> getFunctionVersions() {
+    auto headers = getHeaders();
+    auto versionDataJs = m_restClient.get(
+        fmt::format("https://{}/nvcf/functions/{}", m_baseUrl, m_functionId),
+        "/versions", headers);
+    cudaq::info("Version data: {}", versionDataJs.dump());
+    std::vector<cudaq::NvcfFunctionVersionInfo> versions;
+    versionDataJs["functions"].get_to(versions);
+    return versions;
+  }
+
+public:
+  virtual void setConfig(
+      const std::unordered_map<std::string, std::string> &configs) override {
+    {
+      const auto apiKeyIter = configs.find("api-key");
+      if (apiKeyIter != configs.end())
+        m_apiKey = apiKeyIter->second;
+      if (m_apiKey.empty())
+        throw std::runtime_error("No NVCF API key is provided.");
+    }
+    {
+      const auto funcIdIter = configs.find("function-id");
+      if (funcIdIter != configs.end())
+        m_functionId = funcIdIter->second;
+      if (m_functionId.empty())
+        throw std::runtime_error("No NVCF function Id is provided.");
+    }
+    {
+      auto versions = getFunctionVersions();
+      // Check if a version Id is set
+      const auto versionIdIter = configs.find("version-id");
+      if (versionIdIter != configs.end()) {
+        m_functionVersionId = versionIdIter->second;
+        // Do a sanity check that this is an active version (i.e., usable).
+        const auto versionInfoIter =
+            std::find_if(versions.begin(), versions.end(),
+                         [&](const cudaq::NvcfFunctionVersionInfo &info) {
+                           return info.versionId == m_functionVersionId;
+                         });
+        // Invalid version Id.
+        if (versionInfoIter == versions.end())
+          throw std::runtime_error(
+              fmt::format("Version Id '{}' is not valid for NVCF function Id "
+                          "'{}'. Please check your input.",
```

Comment:
This version Id is user-provided (via command-line, environment variable, or config file), which disables automatic version selection. Hence, the error message was meant to inform users to check their configuration since it is an invalid one. I've changed the wording a bit: `"Please check your NVCF configurations"`. Please let me know if you have a suggestion for the message.

We do echo back the function Id in the error message.